compiler/rustc_expand/src/expand.rs

```diff
@@ -917,7 +917,7 @@ pub fn parse_ast_fragment<'a>(
         }
         AstFragmentKind::Ty => AstFragment::Ty(this.parse_ty()?),
         AstFragmentKind::Pat => {
-            AstFragment::Pat(this.parse_pat_allow_top_alt(None, GateOr::Yes, RecoverComma::No)?)
+            AstFragment::Pat(this.parse_pat_allow_top_alt(None, RecoverComma::No)?)
```

Comment:
I'm a bit confused about this =) does this mean that `pat` accepts `a|b`? I guess I expected some edition logic, and the new fragment matchers, is that in another PR?

Reply:
I didn't fully understand this change very well either, but @petrochenkov seems confident (https://github.com/rust-lang/rust/pull/81869#discussion_r573234448). Apparently, this code is for parsing a pattern produced by a macro expansion.

The code you are probably thinking of is here: https://github.com/rust-lang/rust/pull/79278/files#diff-1a054923fe6e121f46954f52f6660a22970818b5c8b3ef1c0f3b703eae2baef6L122-L128